docker/topic_init.sh

```diff
@@ -1,23 +1,29 @@
 #!/bin/bash
 
+MAX_RETRIES=5
+
 if [ -z NO_VALIDATE ]; then
   radar-schemas-tools validate merged
 fi
 
 # Create topics
 echo "Creating RADAR-base topics..."
 
-if ! radar-schemas-tools create -c "${KAFKA_CONFIG_PATH}" -p $KAFKA_NUM_PARTITIONS -r $KAFKA_NUM_REPLICATION -b $KAFKA_NUM_BROKERS -s "${KAFKA_BOOTSTRAP_SERVERS}" merged; then
-  echo "FAILED TO CREATE TOPICS ... Retrying"
-  if ! radar-schemas-tools create -c "${KAFKA_CONFIG_PATH}" -p $KAFKA_NUM_PARTITIONS -r $KAFKA_NUM_REPLICATION -b $KAFKA_NUM_BROKERS -s "${KAFKA_BOOTSTRAP_SERVERS}" merged; then
-    echo "FAILED TO CREATE TOPICS"
-    exit 1
-  else
-    echo "Created topics at second attempt"
-  fi
-else
-  echo "Topics created."
-fi
+for ((i=1; i<=$MAX_RETRIES; i++)); do
+    if radar-schemas-tools create -c "${KAFKA_CONFIG_PATH}" -p $KAFKA_NUM_PARTITIONS -r $KAFKA_NUM_REPLICATION -b $KAFKA_NUM_BROKERS -s "${KAFKA_BOOTSTRAP_SERVERS}" merged; then
+        echo "Created topics at attempt $i"
+        break
+    else
+        if [ $i -eq $MAX_RETRIES ]; then
+            echo "FAILED TO CREATE TOPICS AFTER $i ATTEMPT(S)"
+            exit 1
+        else
+            echo "FAILED TO CREATE TOPICS ... Retrying"
+            sleep 10
```

Comment:
there is sleep logic within the application too so you can retry immediately too. Alternatively, add `Retrying after waiting for 10 seconds`in the echo to keep users informed that it will wait.

Reply:
Oh good to know. Sleep removed.